### ISSUE TITLE
Generator loss

### ISSUE THREAD

**Issue by pmorerio** [shlizee/Audeo#1]
Hi, looks like the loss for the generator is only this adversarial loss. https://github.com/shlizee/Audeo/blob/962f12450c1956c51eb59bd59b76a1a9dd026d2b/Roll2Midi_train.py#L75

However adversarial loss is MSE in practice
https://github.com/shlizee/Audeo/blob/962f12450c1956c51eb59bd59b76a1a9dd026d2b/Roll2Midi_train.py#L205.

So what is the point of training a discriminator here? 

Thank you,
P.

**Comment by sukun1045** [comment]
Hi, you are right, there should be an adversarial loss as well. I forgot it when I rewrote the code but now I correct it. The idea of Roll2Midi Net is very similar to the [context encoders](https://arxiv.org/pdf/1604.07379.pdf) and we borrow code from its [implementation](https://github.com/eriklindernoren/PyTorch-GAN/blob/master/implementations/context_encoder/context_encoder.py). Thank you.
